### PR TITLE
🎉 Source Google Adwords: support selecting the lookback window for determining conversions

### PR DESCRIPTION
**Pull request by harshithmullapudi**
## What
*Describe what the change is solving*
It helps the user to define the historical replication start date

## How
*Describe the solution*
As the singer already supports it adding it in the schema will let airbyte api configure it

### CODE REVIEW

File: airbyte-integrations/connectors/source-google-adwords-singer/source_google_adwords_singer/spec.json [hunk 1]
```diff
@@ -10,7 +10,8 @@
       "oauth_client_secret",
       "refresh_token",
       "start_date",
-      "customer_ids"
+      "customer_ids",
+      "conversion_window_days"
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-google-adwords-singer/source_google_adwords_singer/spec.json:14]: this should not be a required parameter for backwards compatibility

**Review by sherifnada — COMMENTED** [review]
@harshithmullapudi Thanks for the contribution!! we will also need to pass this input to the singer tap in `source.py`. Would you like to make that change?

File: airbyte-integrations/connectors/source-google-adwords-singer/source_google_adwords_singer/spec.json (hunk 1 above)
Comment (michel-tricot) [reply airbyte-integrations/connectors/source-google-adwords-singer/source_google_adwords_singer/spec.json:14]: Backward compatibility apart, should it be required?

Comment (sherifnada) [reply airbyte-integrations/connectors/source-google-adwords-singer/source_google_adwords_singer/spec.json:14]: nope. It should be optional. It basically decides the lookback window for getting reports. By default this is 30 days, but it should be configurable by the user

File: airbyte-integrations/connectors/source-google-adwords-singer/source_google_adwords_singer/spec.json [hunk 2]
```diff
@@ -39,6 +40,11 @@
         "description": "UTC date and time in the format 2017-01-25T00:00:00Z. Any data before this date will not be replicated.",
         "examples": ["2017-01-25T00:00:00Z"]
       },
+      "conversion_window_days": {
+        "type": "string",
+        "description": "Define the historical replication start date. Change this setting if you want to replicate data beyond Google Ads’s default setting of 30 days.",
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-google-adwords-singer/source_google_adwords_singer/spec.json:45]:
```suggestion
        "description": "Define the historical replication start date. Change this setting if you want to replicate data beyond Google Ads’s default setting of 30 days.",
        "default": "-30",
        "pattern": "^-[0-9]+"
```

File: airbyte-integrations/connectors/source-google-adwords-singer/source_google_adwords_singer/spec.json (hunk 1 above)
Comment (harshithmullapudi) [reply airbyte-integrations/connectors/source-google-adwords-singer/source_google_adwords_singer/spec.json:14]: Yeah I will remove from required

File: airbyte-integrations/connectors/source-google-adwords-singer/source_google_adwords_singer/spec.json (hunk 2 above)
Comment (harshithmullapudi) [reply airbyte-integrations/connectors/source-google-adwords-singer/source_google_adwords_singer/spec.json:45]: Sure

**Comment by harshithmullapudi** [comment]
> @harshithmullapudi Thanks for the contribution!! we will also need to pass this input to the singer tap in `source.py`. Would you like to make that change? 

Yeah will work on that

**Comment by harshithmullapudi** [comment]
> @harshithmullapudi Thanks for the contribution!! we will also need to pass this input to the singer tap in `source.py`. Would you like to make that change?

I think this is already taken care. As this comes from config.json. I did test this.
Screenshot from 2021-04-17 14-21-42

**Review by sherifnada — APPROVED** [review]
@harshithmullapudi looks great, and you are right about `source.py` -- my mistake for the misleading comment. Thanks!

**Comment by sherifnada** [comment]
@harshithmullapudi FYI this is blocked on #2929 -- we're working on that this week and you should be able to access the new connector version once that issue is blocked.

**Comment by michel-tricot** [comment]
Thank you @harshithmullapudi !!!

**Pull request merged**